src/condor_utils/file_transfer.cpp

```diff
@@ -924,27 +923,32 @@ FileTransfer::IsDataflowJob( ClassAd *job_ad ) {
 			oldest_output_timestamp = *output_timestamps.begin();
 			is_dataflow = oldest_output_timestamp > newest_input_timestamp;
 		}
-		// If the executable is more recent than the newest input file, 
+		if(! is_dataflow) { return false; }
+
+		// If the oldest output file is more recent than the executable,
 		// then this is a dataflow job.
 		job_ad->LookupString( ATTR_JOB_CMD, executable_file );
```

Comment:
I guess this works, but I was thinking we'd just move the stat'ing of ATTR_JOB_CMD up after the looping over the input file, and insert the mtime of ATTR_JOB_CMD into `input_timestamps`